internal/services/maintenance/public_maintenance_configurations_data_source.go

```diff
@@ -173,7 +174,7 @@ func dataSourcePublicMaintenanceConfigurationsRead(d *pluginsdk.ResourceData, me
 		return fmt.Errorf("setting `configs`: %+v", err)
 	}
 
-	d.SetId(time.Now().UTC().String())
+	d.SetId(fmt.Sprintf("publicMaintenanceConfigurations/location:%s/scope:%s/recurEvery:%s", locationFilter, scopeFilter, recurEveryFilterRaw))
```

Comment:
```suggestion
	d.SetId(fmt.Sprintf("publicMaintenanceConfigurations;location=%s;scope=%s;recurEvery=%s", locationFilter, scopeFilter, recurEveryFilterRaw))
```

Reply:
this should be using a ARM Style Resource ID (or [base64] encoding this value prior to setting it into the state)